### PR TITLE
MITRE Att&ck - New indicator type "Tactic"

### PR DESCRIPTION
**Pull request by Ni-Knight**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-12180
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11661
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11658
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11657
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11654

## Description
This PR covers a few subjects and issues around MITRE ATT&CK in Cortex TIM:

**Adding MITRE Tactics**
- New Cortex TIM indicator type "Tactic" in CommonTypes.
- Adding a new Tactic indicator class to CSP.
- Adding Tactic as an ObjectName in CSP.
- New MITRE ATT&CK formatter to handle Tactics as well.
- New layout.

**New MITRE ATT&CK Formatter**
- New formatter that formats the MITRE IDs with its name using TIM data and not the command from the feed.
- The formatter now also supports Tactics.

**Updating MITRE ATT&CK Feed to new URL and TAXII 2.1**
- URL adjusted in the feed itself.
- switched the feed to use TAXII 2.1 objects (instead of 2.0).

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CommonScripts/ReleaseNotes/1_15_89.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+#### Scripts
+
+##### New: MITRENameByID_Formatter
+New: Get a MITRE ATT&CK object name by its ID. The script is using TIMs IOCs to find the correct name (MITRE ATT&CK IOCs must exist in the Threat Intel data).
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/ReleaseNotes/1_15_89.md:5]:
```suggestion
New: Get a MITRE ATT&CK object name by its ID. The script is using TIMs IOCs to find the correct name. (MITRE ATT&CK IOCs must exist in the Threat Intel data).
```

File: Packs/CommonScripts/Scripts/MITRENameByIDFormatter/MITRENameByIDFormatter.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+commonfields:
+  id: MITRENameByID_Formatter
+  version: -1
+name: MITRENameByID_Formatter
+script: ''
+type: python
+tags:
+- indicator-format
+comment: Get a MITRE ATT&CK object name by its ID. The script is using TIMs IOCs to find the correct name (MITRE ATT&CK IOCs must exist in the Threat Intel data).
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/MITRENameByIDFormatter/MITRENameByIDFormatter.yml:9]:
```suggestion
comment: Get a MITRE ATT&CK object name by its ID. The script is using TIMs IOCs to find the correct name. (MITRE ATT&CK IOCs must exist in the Threat Intel data).
```

File: Packs/CommonScripts/Scripts/MITRENameByIDFormatter/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+Get a MITRE ATT&CK object name by its ID. The script is using TIMs IOCs to find the correct name (MITRE ATT&CK IOCs must exist in the Threat Intel data).
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/MITRENameByIDFormatter/README.md:1]:
```suggestion
Get a MITRE ATT&CK object name by its ID. The script is using TIMs IOCs to find the correct name. (MITRE ATT&CK IOCs must exist in the Threat Intel data).
```

File: Packs/CommonScripts/Scripts/MITRENameByIDFormatter/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+* MITRE ATT&CK v2
+* mitre-get-indicator-name
+
+## Inputs
+
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| input | The MITRE ATT&amp;CK object ID |
```
Comment (ShirleyDenkberg) [inline Packs/CommonScripts/Scripts/MITRENameByIDFormatter/README.md:26]:
```suggestion
| input | The MITRE ATT&CK object ID. |
```

File: Packs/CommonTypes/ReleaseNotes/3_5_26.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+##### Attack Pattern
+Fixed an issue with Attack Patterns not being added to the context.
+
+##### New: Tactic
+New: New indicator type for MITRE ATT&CK tactics.
+
+#### Layouts
+
+##### New: Tactic Layout
+New: New layout for MITRE ATT&CK tactics<~XSIAM>
```
Comment (ShirleyDenkberg) [inline Packs/CommonTypes/ReleaseNotes/3_5_26.md:13]: Is <~XSIAM> need here?

**Comment by ShirleyDenkberg** [comment]
@DeanArbel @MLainer1 @altmannyarden Doc review completed.

File: Packs/Base/pack_metadata.json [hunk 6]
```diff
@@ -2,7 +2,7 @@
     "name": "Base",
     "description": "The base pack for Cortex XSOAR.",
     "support": "xsoar",
-    "currentVersion": "1.34.47",
+    "currentVersion": "1.34.48",
```
Comment (DeanArbel) [inline Packs/Base/pack_metadata.json:5]: You should make it a minor bump.

File: Packs/CommonScripts/pack_metadata.json [hunk 7]
```diff
@@ -2,7 +2,7 @@
     "name": "Common Scripts",
     "description": "Frequently used scripts pack.",
     "support": "xsoar",
-    "currentVersion": "1.15.88",
+    "currentVersion": "1.15.89",
```
Comment (DeanArbel) [inline Packs/CommonScripts/pack_metadata.json:5]: You should make it a minor bump.

File: Packs/CommonTypes/pack_metadata.json [hunk 8]
```diff
@@ -2,7 +2,7 @@
     "name": "Common Types",
     "description": "This Content Pack will get you up and running in no-time and provide you with the most commonly used incident & indicator fields and types.",
     "support": "xsoar",
-    "currentVersion": "3.5.25",
+    "currentVersion": "3.5.26",
```
Comment (DeanArbel) [inline Packs/CommonTypes/pack_metadata.json:5]: You should make it a minor bump.

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py [hunk 9]
```diff
@@ -6,7 +6,7 @@
 import json
 import urllib3
 from stix2 import TAXIICollectionSource, Filter
-from taxii2client.v20 import Server, Collection, ApiRoot
+from taxii2client.v21 import Server, Collection, ApiRoot
```
Comment (DeanArbel) [inline Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:9]: Nice

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py [hunk 10]
```diff
@@ -115,85 +118,123 @@ def create_indicator(self, item_type, value, mitre_item_json):
...
+                        if item_type.lower() == 'relationship':
                             if mitre_item_json.get('relationship_type') == 'revoked-by':
                                 continue
                             mitre_relationships_list.append(mitre_item_json)
 
                         else:
                             if is_indicator_deprecated_or_revoked(mitre_item_json):
                                 continue
-                            id_to_name[mitre_item_json.get('id')] = value
+                            id_to_name[mitre_item_json['id']] = value
```
Comment (DeanArbel) [inline Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:219]: Why did you replace .get() with []?

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py [hunk 11]
```diff
@@ -115,85 +118,123 @@ def create_indicator(self, item_type, value, mitre_item_json):
...
                             indicator_obj = self.create_indicator(item_type, value, mitre_item_json)
                             add_obj_to_mitre_id_to_mitre_name(mitre_id_to_mitre_name, mitre_item_json)
                             indicators.append(indicator_obj)
                             counter += 1
-                        mitre_id_list.add(mitre_item_json.get('id'))
+                        mitre_id_list.add(mitre_item_json['id'])
```
Comment (DeanArbel) [inline Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:224]: Why did you replace .get() with []?

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py [hunk 12]
```diff
@@ -446,7 +510,7 @@ def get_indicators_command(client, args):
             'Type': entryTypes['note'],
             'Contents': indicators,
             'ContentsFormat': formats['json'],
-            'HumanReadable': tableToMarkdown('MITRE ATT&CK v2 Indicators:', indicators, ['value', 'score', 'type']),
+            'HumanReadable': tableToMarkdown('MITRE ATT&CK v2 Indicators:', indicators[0], ['value', 'score', 'type']),
```
Comment (DeanArbel) [inline Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:513]: Why just the first item?

File: Packs/FeedMitreAttackv2/pack_metadata.json [hunk 13]
```diff
@@ -2,7 +2,7 @@
     "name": "MITRE ATT&CK",
     "description": "Fetches indicators from MITRE ATT&CK.",
     "support": "xsoar",
-    "currentVersion": "1.1.43",
+    "currentVersion": "1.1.44",
```
Comment (DeanArbel) [inline Packs/FeedMitreAttackv2/pack_metadata.json:5]: You should make it a minor bump.

**Review by DeanArbel — APPROVED** [review]
Very nice, just some small comments.

File: Packs/Base/pack_metadata.json (hunk 6 above)
Comment (Ni-Knight) [reply Packs/Base/pack_metadata.json:5]: Roger that

File: Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py (hunk 12 above)
Comment (Ni-Knight) [reply Packs/FeedMitreAttackv2/Integrations/FeedMitreAttackv2/FeedMitreAttackv2.py:513]: the dictionary comes back in index 0
image

**Comment by kobymeir** [comment]
### This PR was automatically updated by a [GitHub Action](https://github.com/demisto/content/actions/runs/12122003002)
- **CommonScripts** pack version was bumped to **1.16.0**.
- **Base** pack version was bumped to **1.37.0**.

To stop automatic version bumps, add the `ignore-auto-bump-version` label to the github PR.

**Comment by kobymeir** [comment]
### This PR was automatically updated by a [GitHub Action](https://github.com/demisto/content/actions/runs/12139309775)
- **CommonScripts** pack version was bumped to **1.17.0**.

To stop automatic version bumps, add the `ignore-auto-bump-version` label to the github PR.

**Comment by kobymeir** [comment]
### This PR was automatically updated by a [GitHub Action](https://github.com/demisto/content/actions/runs/12140004404)
- **CommonScripts** pack version was bumped to **1.18.0**.

To stop automatic version bumps, add the `ignore-auto-bump-version` label to the github PR.

**Comment by Ni-Knight** [comment]
This PR added a toVersion to the Attack Pattern as we switched its formatter in the new version. As such it requires a force-merge

**Comment by kobymeir** [comment]
### This PR was automatically updated by a [GitHub Action](https://github.com/demisto/content/actions/runs/12222018616)
- **CommonScripts** pack version was bumped to **1.18.0**.

To stop automatic version bumps, add the `ignore-auto-bump-version` label to the github PR.

**Pull request merged**